app/src/main/java/su/levenetc/android/textsurface/sample/animations/complex.kt

```diff
@@ -0,0 +1,116 @@
+import su.levenetc.android.textsurface.animations.effects.reveal.SideCut
+import su.levenetc.android.textsurface.animations.generic.Delay
+import su.levenetc.android.textsurface.animations.sets.Parallel
+import su.levenetc.android.textsurface.animations.sets.Sequential
+import su.levenetc.android.textsurface.constants.Align
+import su.levenetc.android.textsurface.constants.Direction
+import su.levenetc.android.textsurface.constants.Pivot
+import su.levenetc.android.textsurface.constants.Side
+
+fun playCookieThumper(textSurface: TextSurface, assetManager: AssetManager) {
+
+    val paint = Paint().apply {
+        isAntiAlias = true
+        typeface = Typeface.createFromAsset(assetManager, "fonts/Roboto-Black.ttf")
+    }
+
+    val textDaai = Text.Builder("Daai")
+            .setPaint(paint)
+            .setSize(64f)
+            .setAlpha(0)
+            .setColor(Color.WHITE)
+            .setPosition(Align.SURFACE_CENTER).build()
+    val textBraAnies = Text.Builder("bra Anies")
+            .setPaint(paint)
+            .setSize(44f)
+            .setAlpha(0)
+            .setColor(Color.RED)
+            .setPosition(Align.BOTTOM_OF, textDaai).build()
+    val textFokkenGamBra = Text.Builder(" hy's n fokken gam bra.")
+            .setPaint(paint)
+            .setSize(44f)
+            .setAlpha(0)
+            .setColor(Color.RED)
+            .setPosition(Align.RIGHT_OF, textBraAnies).build()
+    val textHaai = Text.Builder("Haai!!")
+            .setPaint(paint)
+            .setSize(74f)
+            .setAlpha(0)
+            .setColor(Color.RED)
+            .setPosition(Align.BOTTOM_OF, textFokkenGamBra).build()
+    val textDaaiAnies = Text.Builder("Daai Anies")
+            .setPaint(paint)
+            .setSize(44f)
+            .setAlpha(0)
+            .setColor(Color.WHITE)
+            .setPosition(Align.BOTTOM_OF or Align.CENTER_OF, textHaai).build()
+    val texThyLamInnie = Text.Builder(" hy lam innie mang ja.")
+            .setPaint(paint)
+            .setSize(44f)
+            .setAlpha(0)
+            .setColor(Color.WHITE)
+            .setPosition(Align.RIGHT_OF, textDaaiAnies).build()
+    val textThrowDamn = Text.Builder("Throw damn")
+            .setPaint(paint)
+            .setSize(44f)
+            .setAlpha(0)
+            .setColor(Color.RED)
+            .setPosition(Align.BOTTOM_OF or Align.CENTER_OF, texThyLamInnie).build()
+    val textDevilishGang = Text.Builder("devilish gang")
+            .setPaint(paint)
+            .setSize(44f)
+            .setAlpha(0)
+            .setColor(Color.RED)
+            .setPosition(Align.BOTTOM_OF or Align.CENTER_OF, textThrowDamn).build()
+    val textSignsInTheAir = Text.Builder("signs in the air.")
+            .setPaint(paint)
+            .setSize(44f)
+            .setAlpha(0)
+            .setColor(Color.RED)
+            .setPosition(Align.BOTTOM_OF or Align.CENTER_OF, textDevilishGang).build()
+    textSurface.play(
+            Sequential(
```

Comment:
Seems like a good candidate for type-safe builders https://kotlinlang.org/docs/type-safe-builders.html

Reply:
Oh, damn, true. I either didn't know about type-safe builder or completely forgot about it.
Added builders: https://github.com/elevenetc/TextSurface/blob/f240b0d4e835dc6a9547b98c09fa392349b0b4bb/library/src/main/java/su/levenetc/android/textsurface/animations/builder.kt
And updated animation construction: https://github.com/elevenetc/TextSurface/blob/4ff41a26b604281be7203f801af2af3e3200dfd3/app/src/main/java/su/levenetc/android/textsurface/sample/animations/complex.kt